doc/rm/c_cpp_coding_style.md

```diff
@@ -110,6 +110,20 @@ This convention helps readers distinguish which files they should not expect to
 
 The above rules also do not apply to system includes, which should be included by the names dictated by the ISO standard, e.g. `#include <stddef.h>`.
 
+### Linker Script Provided Definitions
+
+Some C/C++ programs may need to use symbols that are defined by a linker script.
+Referring to linker script symbols can be complex and error-prone, as they don't quite work like C's global variables.
+
+If you need to refer to the linker-script-defined symbol `_my_linker_symbol`, the correct way to do so is with an incomplete extern char array declaration, as shown below.
+This allows you to treat the value as a pointer, without C expecting storage to be allocated for the symbol.
+It is good practice to provide a comment that directs others to where the symbol is actually defined.
+```c
+/**
+ * `_my_linker_symbol` is defined in the linker script `sw/device/my_feature.ld`.
+ */
+extern char _my_linker_symbol[];
+```
```

Comment:
Can you link to https://sourceware.org/binutils/docs/ld/Source-Code-Reference.html, as the description there is actually rather good and complete.

Reply:
If you do so, please make it clear this is part of the binutils ld manual, rather than some random URL.

Reply:
Done